### ISSUE TITLE
Docker 'pull' from private registry ceased to work with "authConfig does not conform to given endpoint ( != <private>"

### ISSUE THREAD

**Issue by kasur** [docker/docker#12117]
~~~
[root@centos7 ~]# uname -a
Linux centos7.kasured.com 3.10.0-229.1.2.el7.x86_64 #1 SMP Fri Mar 27 03:04:26 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

[root@centos7 ~]# docker version
Client version: 1.5.0-dev
Client API version: 1.18
Go version (client): go1.3.3
Git commit (client): fc0329b/1.5.0
OS/Arch (client): linux/amd64
Server version: 1.5.0-dev
Server API version: 1.18
Go version (server): go1.3.3
Git commit (server): fc0329b/1.5.0
OS/Arch (server): linux/amd64

[root@centos7 ~]# docker login <private>
Username: ***
Password: 
Email: *****
Login Succeeded

[root@centos7 ~]# docker pull <private>/<user>/redis
Trying to pull repository <private>/<user>/redis ... failed
FATA[0000] Authentication is required.

[root@centos7 ~]# systemctl status docker.service -l
docker.service - Docker Application Container Engine
   Loaded: loaded (/usr/lib/systemd/system/docker.service; enabled)
   Active: active (running) since Mon 2015-04-06 20:49:51 MSK; 11min ago
     Docs: http://docs.docker.com
 Main PID: 10923 (docker)
   CGroup: /system.slice/docker.service
           └─10923 /usr/bin/docker -d --selinux-enabled
Apr 06 21:00:14 centos7.kasured.com docker[10923]: time="2015-04-06T21:00:14+03:00" level="info" msg="-job resolve_repository(<private>/<user>/redis) = OK (0)"
Apr 06 21:00:14 centos7.kasured.com docker[10923]: time="2015-04-06T21:00:14+03:00" level="info" msg="authConfig does not conform to given endpoint ( != <private>)"
Apr 06 21:00:14 centos7.kasured.com docker[10923]: Authentication is required.
Apr 06 21:00:14 centos7.kasured.com docker[10923]: time="2015-04-06T21:00:14+03:00" level="info" msg="-job pull(<private>/<user>/redis, latest) = ERR (1)"
~~~

**Comment by dmcgowan** [comment]
What is the commit ID of master you are building off? I do not see a commit associated with "fc0329b" in the tree. I do not see the does not conform message referenced anywhere in the master code base, do you have code additions you are testing or using a fork?

Also you blacked out `</private>` which is fine, but can you let me know what the format is that you are using?  For example for login are you doing https://private.example.com?

**Comment by kasur** [comment]
Hola, Derek,

1) I suppose this is the commit reference https://github.com/docker/docker/commit/fc0329baa1cc2f73349d564fb3d32e0112c51385 (auto-inserted in edit mode but it might as well be a collision)

2) The docker version that I am using comes from default extras repo for centos7
Version     : 1.5.0
Release     : 28.el7.centos
Repo        : extras/7/x86_64 

3) I am using a private registry and making the reference docker-reg.[private].com

4) The flow stopped working after the CentOS upgrade 7.0->7.1

--Evgeny

**Comment by dmcgowan** [comment]
The error you are seeing is not part of the Docker git repository (although Github kindly references it for you under our repository) but from an individual fork.  The change introducing this error message is at https://github.com/docker/docker/commit/dc670b83835d4a897ae8ae42d60c721d5b3031ab.  Since this change is not related to any committed code or security patch, I would suggest reaching out to the author of this commit @miminar or the centos package creater @lsm5 for a solution.

**Comment by miminar** [comment]
Yep, this is rhel specific problem. @kasur, let's please continue discussion in [this bugzilla](https://bugzilla.redhat.com/show_bug.cgi?id=1209439).

**Comment by kasur** [comment]
I have also submitted the issue here http://bugs.centos.org/view.php?id=8406.

**Comment by kasured** [comment]
For the sake of consistency I would like to mention that the issue 
https://bugzilla.redhat.com/show_bug.cgi?id=1209439 has been marked as fixed and is no longer reproducible with the latest available build of docker which is 1.6.0-1.el7

Please see more details on the bugzilla issue page.

--Evgeny
